cwa/olin/src/lib.rs

```diff
@@ -1,54 +1,92 @@
+/// Main classes to perform the low-level operations
+/// described in the Olin specs. In particular the basic
+/// "actions": open, read, write, close, flush on files.
+/// 
+/// Modules:
+/// * sys
+/// * err
+/// * log
+/// * env
+/// * runtime
+/// * startup
+/// * time
+/// * stdio
+/// * random
+///
+/// ```Resource``` class defines a generic class for a resource
+/// with methods bound for FFI access
+
 extern crate chrono;
 
 use std::io::{self, Read, Write};
 
+/// import HTTP client from local
+/// TODO: couldn't get how this is used as it seems not to be called anywhere
 pub mod http;
+/// import panic handling from local
 pub mod panic;
 
+/// system-level operations, interface and FFI access:
+/// * setup and low-level operations
+/// * bindings to environment variables
+/// * bindings for Go runtime
```

Comment:
The fact that the runtime is in Go is an implementation detail a layer above this code. It is not relevant here.

Reply:
At this point it is relevant as we are trying to document for ourselves and newcomers. Once everything is clearer this can be omitted (for now I put it in parenthesis)